app/repositories/DepartureRepository.scala

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import config.AppConfig
+import javax.inject.Inject
+import models.Departure
+import models.DepartureId
+import models.DepartureStatus
+import models.Message
+import models.MessageStatus
+import models.MongoDateTimeFormats
+import models.MovementReferenceNumber
+import play.api.libs.json.Json
+import play.modules.reactivemongo.ReactiveMongoApi
+import reactivemongo.api.commands.WriteResult
+import reactivemongo.bson.BSONDocument
+import reactivemongo.api.bson.collection.BSONSerializationPack
+import reactivemongo.api.indexes.Index.Aux
+import reactivemongo.api.indexes.IndexType
+import reactivemongo.play.json.collection.JSONCollection
+import utils.IndexUtils
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+class DepartureRepository @Inject()(mongo: ReactiveMongoApi, appConfig: AppConfig)(implicit ec: ExecutionContext) extends MongoDateTimeFormats {
```

Comment:
You aren't using most of this class. Get rid of what you won't need. Especially since my understanding is you won't be editing the models directly. That way you don't leave the door open to unlocked editing when someone inherits the project.

Reply:
I tried this but the tests for the get operation make use of insert which make use of the setter methods. I could only get rid of the files that aren't being used.

Reply:
I mean, fair for the insert - but you don't need the addNewResponseMessage or any of the others. I don't think you even need find by ref and eori. Find what you don't need and clean it up.